src/App.tsx

```diff
@@ -1,61 +1,113 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { Todo, FullTodo } from './react-app-env';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+const fullTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState(fullTodos);
+  const [user, setUser] = useState(0);
+  const [title, setTitle] = useState('');
+  const [submit, setSubmit] = useState(false);
+
+  function biggestID() {
+    let biggest = 0;
+
+    todos.forEach(todo => {
+      if (todo.id > biggest) {
+        biggest = todo.id;
+      }
+    });
+
+    return biggest + 1;
+  }
+
+  const addTodo = () => {
+    const newTodo: FullTodo = {
+      id: biggestID(),
+      title,
+      completed: false,
+      userId: +user - 1,
+      user: usersFromServer[user - 1],
```

Comment:
```suggestion
      user: findUser(userId),
```
Better store userId (not index) and search for correct user. Order can be changed, id - not